packages/build-utils/test/unit.test.ts

```diff
@@ -413,3 +414,22 @@ it('should detect pnpm Workspaces', async () => {
   expect(result.cliType).toEqual('pnpm');
   expect(result.lockfileVersion).toEqual(5.3);
 });
+
+it('should only invoke `runNpmInstall()` once per `package.json` file', async () => {
+  const meta: Meta = {};
+  const fixture = path.join(__dirname, 'fixtures', '02-zero-config-api');
+  const apiDir = path.join(fixture, 'api');
+  const [run1, run2, run3] = await Promise.all([
+    runNpmInstall(apiDir, [], undefined, meta),
+    runNpmInstall(apiDir, [], undefined, meta),
+    runNpmInstall(fixture, [], undefined, meta),
+  ]);
```

Comment:
Does it work if invoked sequentially (as opposed to concurrently)?

Reply:
Yes. Added a test.